### PR TITLE
feat: set a block to be called when complete initialize

### PR DESCRIPTION
**Pull request by yuhao900914**
### Checklist

* [ ] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-iOS/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:

### CODE REVIEW

**Comment by dantetam** [comment]
LGTM! Pretty straightforward to me

**Review by dantetam — APPROVED** [review]

File: Sources/Amplitude/Public/Amplitude.h [hunk 1]
```diff
@@ -30,7 +30,7 @@ NS_ASSUME_NONNULL_BEGIN
 
 typedef NSString *_Nonnull (^AMPAdSupportBlock)(void);
 typedef NSDictionary *_Nullable (^AMPLocationInfoBlock)(void);
-
+typedef void (^AMPInitializeCompleteSupportBlock)(void);
```
Comment (haoliu-amp) [inline Sources/Amplitude/Public/Amplitude.h:33]: The name is too long. How about `AMPInitCompletionBlock`?

File: Sources/Amplitude/Public/Amplitude.h [hunk 2]
```diff
@@ -181,6 +181,17 @@ typedef NSDictionary *_Nullable (^AMPLocationInfoBlock)(void);
...
+/**
+ * Sets a block to be called after completely initialized.
+ *
+ * Example:
+ *  __typeof(amp) __weak weakAmp = amp;
+ *  amp.initializeSupportBlock = ^(void){
+ *     NSLog(@"deviceId: %@, userId: %@", weakAmp.deviceId, weakAmp.userId);
+ *  };
+ */
+@property (nonatomic, strong, nullable) AMPInitializeCompleteSupportBlock initializeCompleteSupportBlock;
```
Comment (haoliu-amp) [inline Sources/Amplitude/Public/Amplitude.h:193]: Same, please rename it.

File: Sources/Amplitude/Public/Amplitude.h [hunk 3]
```diff
@@ -181,6 +181,17 @@ typedef NSDictionary *_Nullable (^AMPLocationInfoBlock)(void);
  */
 @property (nonatomic, copy, readonly) NSString *contentTypeHeader;
 
+/**
+ * Sets a block to be called after completely initialized.
+ *
+ * Example:
+ *  __typeof(amp) __weak weakAmp = amp;
+ *  amp.initializeSupportBlock = ^(void){
```
Comment (haoliu-amp) [inline Sources/Amplitude/Public/Amplitude.h:189]: The name in the example doesn't match what you have in the code.

**Review by haoliu-amp — CHANGES_REQUESTED** [review]

File: Sources/Amplitude/Public/Amplitude.h (hunk 1 above)
Comment (yuhao900914) [reply Sources/Amplitude/Public/Amplitude.h:33]: Much better! Will update the name to this.

**Review by haoliu-amp — APPROVED** [review]

**Review by qingzhuozhen — APPROVED** [review]

**Review by bgiori — APPROVED** [review]
🚀

**Review by qingzhuozhen — APPROVED** [review]

**Pull request merged**